### ISSUE TITLE
bug: SNS to SQS subscription breaks on republished 0.13.3 version

### ISSUE THREAD

**Issue by jansepke** [localstack/localstack#5337]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When a SNS to SQS subscription is used it will throw an exception in the latest version.

```
2022-01-24T05:57:14.420:INFO:localstack.services.sns.sns_listener: Unable to forward SNS message to SQS: Could not connect to the endpoint URL: "http://localhost:4567/" Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/util/connection.py", line 95, in create_connection
    raise err
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/util/connection.py", line 85, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/httpsession.py", line 385, in send
    urllib_response = conn.urlopen(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/util/retry.py", line 525, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/packages/six.py", line 770, in reraise
    raise value
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 398, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 239, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.8/http/client.py", line 1256, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/awsrequest.py", line 92, in _send_request
    rval = super(AWSConnection, self)._send_request(
  File "/usr/local/lib/python3.8/http/client.py", line 1302, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.8/http/client.py", line 1251, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/awsrequest.py", line 120, in _send_output
    self.send(msg)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/awsrequest.py", line 204, in send
    return super(AWSConnection, self).send(str)
  File "/usr/local/lib/python3.8/http/client.py", line 951, in send
    self.connect()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 205, in connect
    conn = self._new_conn()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <botocore.awsrequest.AWSHTTPConnection object at 0x7f0254f56040>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/code/localstack/localstack/services/sns/sns_listener.py", line 522, in message_to_subscriber
    sqs_client.send_message(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 391, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 705, in _make_api_call
    http, parsed_response = self._make_request(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 725, in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 104, in make_request
    return self._send_request(request_dict, operation_model)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 138, in _send_request
    while self._needs_retry(attempts, operation_model, request_dict,
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 254, in _needs_retry
    responses = self._event_emitter.emit(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/hooks.py", line 357, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/hooks.py", line 228, in emit
    return self._emit(event_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/hooks.py", line 211, in _emit
    response = handler(**kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 183, in __call__
    if self._checker(attempts, response, caught_exception):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 250, in __call__
    should_retry = self._should_retry(attempt_number, response,
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 277, in _should_retry
    return self._checker(attempt_number, response, caught_exception)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 316, in __call__
    checker_response = checker(attempt_number, response,
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 222, in __call__
    return self._check_caught_exception(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 359, in _check_caught_exception
    raise caught_exception
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 201, in _do_get_response
    http_response = self._send(request)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 270, in _send
    return self.http_session.send(request)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/httpsession.py", line 414, in send
    raise EndpointConnectionError(endpoint_url=request.url, error=e)
botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "http://localhost:4567/"

2022-01-24T05:57:17.918:INFO:localstack.services.sns.sns_listener: Unable to forward SNS message to SQS: Could not connect to the endpoint URL: "http://localhost:4567/" Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/util/connection.py", line 95, in create_connection
    raise err
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/util/connection.py", line 85, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/httpsession.py", line 385, in send
    urllib_response = conn.urlopen(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/util/retry.py", line 525, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/packages/six.py", line 770, in reraise
    raise value
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 398, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 239, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.8/http/client.py", line 1256, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/awsrequest.py", line 92, in _send_request
    rval = super(AWSConnection, self)._send_request(
  File "/usr/local/lib/python3.8/http/client.py", line 1302, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.8/http/client.py", line 1251, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/awsrequest.py", line 120, in _send_output
    self.send(msg)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/awsrequest.py", line 204, in send
    return super(AWSConnection, self).send(str)
  File "/usr/local/lib/python3.8/http/client.py", line 951, in send
    self.connect()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 205, in connect
    conn = self._new_conn()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <botocore.awsrequest.AWSHTTPConnection object at 0x7f0254583100>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/code/localstack/localstack/services/sns/sns_listener.py", line 522, in message_to_subscriber
    sqs_client.send_message(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 391, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 705, in _make_api_call
    http, parsed_response = self._make_request(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 725, in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 104, in make_request
    return self._send_request(request_dict, operation_model)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 138, in _send_request
    while self._needs_retry(attempts, operation_model, request_dict,
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 254, in _needs_retry
    responses = self._event_emitter.emit(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/hooks.py", line 357, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/hooks.py", line 228, in emit
    return self._emit(event_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/hooks.py", line 211, in _emit
    response = handler(**kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 183, in __call__
    if self._checker(attempts, response, caught_exception):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 250, in __call__
    should_retry = self._should_retry(attempt_number, response,
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 277, in _should_retry
    return self._checker(attempt_number, response, caught_exception)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 316, in __call__
    checker_response = checker(attempt_number, response,
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 222, in __call__
    return self._check_caught_exception(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 359, in _check_caught_exception
    raise caught_exception
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 201, in _do_get_response
    http_response = self._send(request)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 270, in _send
    return self.http_session.send(request)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/httpsession.py", line 414, in send
    raise EndpointConnectionError(endpoint_url=request.url, error=e)
botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "http://localhost:4567/"

2022-01-24T05:57:18.259:INFO:localstack.services.sns.sns_listener: Unable to forward SNS message to SQS: Could not connect to the endpoint URL: "http://localhost:4567/" Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/util/connection.py", line 95, in create_connection
    raise err
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/util/connection.py", line 85, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/httpsession.py", line 385, in send
    urllib_response = conn.urlopen(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/util/retry.py", line 525, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/packages/six.py", line 770, in reraise
    raise value
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 398, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 239, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.8/http/client.py", line 1256, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/awsrequest.py", line 92, in _send_request
    rval = super(AWSConnection, self)._send_request(
  File "/usr/local/lib/python3.8/http/client.py", line 1302, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.8/http/client.py", line 1251, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/awsrequest.py", line 120, in _send_output
    self.send(msg)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/awsrequest.py", line 204, in send
    return super(AWSConnection, self).send(str)
  File "/usr/local/lib/python3.8/http/client.py", line 951, in send
    self.connect()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 205, in connect
    conn = self._new_conn()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <botocore.awsrequest.AWSHTTPConnection object at 0x7f025456dca0>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/code/localstack/localstack/services/sns/sns_listener.py", line 522, in message_to_subscriber
    sqs_client.send_message(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 391, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 705, in _make_api_call
    http, parsed_response = self._make_request(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 725, in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 104, in make_request
    return self._send_request(request_dict, operation_model)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 138, in _send_request
    while self._needs_retry(attempts, operation_model, request_dict,
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 254, in _needs_retry
    responses = self._event_emitter.emit(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/hooks.py", line 357, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/hooks.py", line 228, in emit
    return self._emit(event_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/hooks.py", line 211, in _emit
    response = handler(**kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 183, in __call__
    if self._checker(attempts, response, caught_exception):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 250, in __call__
    should_retry = self._should_retry(attempt_number, response,
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 277, in _should_retry
    return self._checker(attempt_number, response, caught_exception)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 316, in __call__
    checker_response = checker(attempt_number, response,
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 222, in __call__
    return self._check_caught_exception(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/retryhandler.py", line 359, in _check_caught_exception
    raise caught_exception
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 201, in _do_get_response
    http_response = self._send(request)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 270, in _send
    return self.http_session.send(request)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/httpsession.py", line 414, in send
    raise EndpointConnectionError(endpoint_url=request.url, error=e)
botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "http://localhost:4567/"
```

### Expected Behavior

SNS to SQS subscription should work as before version 0.13.3

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

docker-compose.yml:
```
version: '3.5'
services:
  localstack:
    image: localstack/localstack:0.13.3
    ports:
      - "4567:4566"
    volumes:
      - ./localstack/init-localstack.sh:/docker-entrypoint-initaws.d/init-localstack.sh
    environment:
      - DEFAULT_REGION=eu-central-1
      - SERVICES=sns,sqs,dynamodb,s3
      - SQS_PORT_EXTERNAL=4567
      - START_WEB=0
```

init-localstack.sh:
```sh
awslocal sns create-topic --name test-output-topic

awslocal sqs create-queue --queue-name test-output-queue

awslocal sns subscribe \
      --topic-arn arn:aws:sns:eu-central-1:000000000000:test-output-topic \
      --protocol sqs \
      --attributes RawMessageDelivery=true \
      --notification-endpoint http://localhost:4567/000000000000/test-output-queue
```

send a message:
`docker compose exec localstack awslocal sns publish --topic-arn arn:aws:sns:eu-central-1:000000000000:test-output-topic --message test`
will return a MessageId but the docker compose log will show the above exception

### Environment

```markdown
- OS: Ubuntu 21.10
- LocalStack: 0.13.3
```

### Anything else?

Bug was likely introduced by 31199e5bd9d0b61dc14e2e8f256b409ecc5ccc20

**Comment by whummer** [comment]
Thanks @jansepke . Can you please share your full environment configuration (e.g., docker-compose.yml file) - looks like you're using `4567` (instead of the default edge port `4566`), which may have an impact on the processing logic. Thanks!

**Comment by jansepke** [comment]
@whummer the complete docker-compose.yml is at the end of the bug report. we are just using docker port mapping to expose the default edge port 4566 as port 4567 and configure `SQS_PORT_EXTERNAL=4567` for this to work.

**Comment by jansepke** [comment]
fixed in v0.14.0
